### DISCUSSION TITLE
Pulsar Java Client 4.0.0 - new Warn message "can't add message to the container"

### DISCUSSION THREAD

**Discussion by peterh-wob** [apache/pulsar#23571] [Q&A] [answered]
Hey Everyone!
I've recently updated to 4.0.0 on both the server and client side. I'm using the java client, and after the update I noticed a new warn message:

> WARN  o.a.pulsar.client.impl.MessagesImpl - can't add message to the container, has exceeded the maxNumberOfMessages : 500

[source](https://github.com/apache/pulsar/blob/c266db236df5f57bf59df1b25a23521419b2ad02/pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessagesImpl.java#L54) | [gh issue](https://github.com/apache/pulsar/issues/21631)
Doesn't seem to cause any error, but I couldn't find any explanation or solution to this.
Should I worry about it? How should I configure the consumer to get rid of these warns?

**Comment by peterh-wob** [comment, marked as answer]
@alpreu already answered the question on slack, I'll copy it here in case of someone needs it! Thanks Alexander!

> It’s not an issue, just a noisy log. In case you want to change the size of the maximum number of messages you can do so using the BatchReceivePolicy on the Consumer, you won’t get rid of the log though

**Reply by lhotari** [reply]
I also asked the PR author @Technoboy- in https://github.com/apache/pulsar/pull/21924#issuecomment-2461847397.
